### PR TITLE
Added Support for Users and Permissions GET/POST URLs

### PR DESCRIPTION
**Pull request by jitendra-xoriant**
Added support for Following Users and Permissions URLs
GET /users
GET /users/{userId}
GET /users/{userId}/permissions
POST /users
POST /users/{userId}/permissions

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/models/Deal.java [hunk 1]
```diff
@@ -337,9 +337,9 @@ public Form getForm() {
 		if (this.getPermissions() != null) {
 			Map<String, Object> pMap = new HashMap<String, Object>();
 			pMap.put("advertiser_ids", this.getPermissions().getAdvertiser_ids());
-			pMap.put("agency_ids", this.getPermissions().getAgency_ids());
-			pMap.put("organization_ids", this.getPermissions().getOrganization_ids());
-			pMap.put("all_organizations", this.getPermissions().isAll_organizations());
+			pMap.put("agency_ids", this.getPermissions().getAgencyIds());
+			pMap.put("organization_ids", this.getPermissions().getOrganizationIds());
+			pMap.put("all_organizations", this.getPermissions().isAllOrganizations());
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Deal.java]: good job.

File: src/main/java/com/mediamath/terminalone/models/Permissions.java [hunk 2]
```diff
@@ -54,8 +78,24 @@ public void setAgency_ids(List<Integer> agency_ids) {
...
+	}
+
+	public List<Access> getFlags() {
+		return flags;
+	}
+
+	public void setFlags(List<Access> flags) {
+		this.flags = flags;
+	}
+
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Permissions.java]: Looks good to me.

File: src/main/java/com/mediamath/terminalone/models/User.java [hunk 3]
```diff
@@ -322,14 +330,106 @@ public void setViewSegments(boolean view_segments) {
     this.view_segments = view_segments;
   }
 
+	public boolean isLinkSaml() {
+		return link_saml;
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/User.java]: whats this used for?

Comment (jitendra-xoriant) [reply src/main/java/com/mediamath/terminalone/models/User.java]: some additional fiedls are coming in response, so added them in our entity like name, link_saml, email

**Pull request merged**